### PR TITLE
S8/feature/handle no response from ipinfo

### PR DESCRIPTION
**Pull request by acrollet**
You'll want to add a row to your settings table:

name: collectUserIp
value: (1 or 0)
type: boolean

### CODE REVIEW

File: resources/assets/js/components/OpenDialogChat.vue [hunk 1]
```diff
@@ -387,6 +382,37 @@ export default {
         }, 200);
       });
     },
+    getUserIp() {
```
Comment (stu-greenshoots) [inline resources/assets/js/components/OpenDialogChat.vue]: This happens outside of the flow of init settings now, does it make sense for it to happen at the same time?

File: resources/assets/js/components/OpenDialogChat.vue [hunk 2]
```diff
@@ -387,6 +382,37 @@ export default {
...
+      axios.get('https://ipinfo.io/').then(
+        (response) => {
+          const browserInfo = detect();
+
+          const ipAddress = response.data.ip;
```
Comment (stu-greenshoots) [inline resources/assets/js/components/OpenDialogChat.vue]: This only needs to collect the IP and country data, not everything else

File: resources/assets/js/components/OpenDialogChat.vue [hunk 3]
```diff
@@ -387,6 +382,37 @@ export default {
...
+          const { country } = response.data;
+          const browserLanguage = navigator.language || navigator.userLanguage;
+          const { os } = browserInfo;
+          const browser = `${browserInfo.name} ${browserInfo.version}`;
+          const timezone = jstz.determine().name();
+
+          this.userInfo = {
```
Comment (stu-greenshoots) [inline resources/assets/js/components/OpenDialogChat.vue]: Could re-setting the userInfo (after initSetting set ip to 'n\a' cause any inconsistencies?

File: resources/assets/js/components/OpenDialogChat.vue [hunk 4]
```diff
@@ -541,6 +567,10 @@ export default {
         this.useAvatars = config.useAvatars;
       }
 
+      if (Object.prototype.hasOwnProperty.call(config, 'collectUserIp')) {
```
Comment (stu-greenshoots) [inline resources/assets/js/components/OpenDialogChat.vue]: can you explain this to me?

**Comment by stu-greenshoots** [comment]
> You'll want to add a row to your settings table:
> 
> name: collectUserIp
> value: (1 or 0)
> type: boolean

Can you add this to the webcaht settings set up script in the BDO project please

**Comment by acrollet** [comment]
@stu-greenshoots PR for BDO: https://github.com/GreenShootLabs/BDO-OpenDialog/pull/30

File: resources/assets/js/components/OpenDialogChat.vue (hunk 4 above)
Comment (acrollet) [reply resources/assets/js/components/OpenDialogChat.vue]: I was concerned that a `if (config.collectUserIp)` test would not pass when it was set to false, but I think that was a condition brought on by an excess of php development. I've updated...

File: resources/assets/js/components/OpenDialogChat.vue (hunk 1 above)
Comment (acrollet) [reply resources/assets/js/components/OpenDialogChat.vue]: Sorry, how do you mean? We do need to wait until the settings have been received so we don't try to get the IP if collection is disabled...

File: resources/assets/js/components/OpenDialogChat.vue (hunk 4 above)
Comment (acrollet) [reply resources/assets/js/components/OpenDialogChat.vue]: argh, I was actually right the first time - I was testing with useAvatars, but its default is false so the conditional works. We should probably use the object property check for all boolean config settings.

File: resources/assets/js/components/OpenDialogChat.vue (hunk 3 above)
Comment (acrollet) [reply resources/assets/js/components/OpenDialogChat.vue]: I don't think so, the chat is always initialised after the call to get the IP completes. We could enforce that if you want, at the risk of slowing down/breaking the widget load if the ipinfo call is slow or doesn't succeed.

File: resources/assets/js/components/OpenDialogChat.vue (hunk 2 above)
Comment (acrollet) [reply resources/assets/js/components/OpenDialogChat.vue]: 👍

File: resources/assets/js/components/OpenDialogChat.vue (hunk 4 above)
Comment (stu-greenshoots) [reply resources/assets/js/components/OpenDialogChat.vue]: :) makes sense!

File: resources/assets/js/components/OpenDialogChat.vue (hunk 3 above)
Comment (stu-greenshoots) [reply resources/assets/js/components/OpenDialogChat.vue]: ok, all good!

File: resources/assets/js/components/OpenDialogChat.vue (hunk 1 above)
Comment (stu-greenshoots) [reply resources/assets/js/components/OpenDialogChat.vue]: valid point...

**Pull request merged**